p2p/dialer.go

```diff
@@ -174,6 +187,16 @@ func (m *dialer) exchangeConfig(consumerID, providerID identity.Identity, servic
 		return nil, fmt.Errorf("could not send signed msg: %v", err)
 	}
 
+	// TODO: put this under feature set check when its available
+	// wait until provider confirms that channel handlers are ready
+	select {
+	case <- ready:
+		log.Debug().Msg("Received handlers ready message from provider")
+	// NOTE: We can make this timeout much larger when all providers migrate to handlers ready message
+	case <- time.After(1*time.Second):
```

Comment:
It can be made larger now, providers are not using it yet. There is one problem. Dial allows to pass timeout so from api perspective if I call

```go
Dial(...., 10 * time.Second)
``` 
I expect that method call to Dial will take no longer that this time.

It can be solved by instead passing context

```go
ctx, cancel := context.WithTimeout(context.Background(), 10 * time.Second)
Dial(ctx) // Internally pass context to method calls to nats
```

But need to check if nats allows to pass context to Request method.

Reply:
Modified Dial to accept timeout context. It is what you imagine should be?